cmd/terramate/cli/script_run.go

```diff
@@ -88,63 +88,95 @@ func (c *cli) runScript() {
 				logger.Fatal().Err(err).Msg("failed to eval script")
 			}
 
-			for jobNum, j := range evalScript.Jobs {
-				c.output.MsgStdOut("")
-				for cmdNum, cmd := range j.Commands() {
-					printCommand(c.stderr, cmd.Args, st.Dir().String(), jobNum, cmdNum)
-
-					env, err := run.LoadEnv(c.cfg(), st.Stack)
-					if err != nil {
-						logger.Fatal().Err(err).Msg("failed to load env")
+			for jobIdx, job := range evalScript.Jobs {
+				for cmdIdx, cmd := range job.Commands() {
+					exc := RunContext{
+						Stack:        st.Stack,
+						Cmd:          cmd.Args,
+						ScriptIdx:    scriptIdx,
+						ScriptJobIdx: jobIdx,
+						ScriptCmdIdx: cmdIdx,
 					}
 
-					if err := c.executeCommand(cmd.Args, st.Dir().HostPath(c.rootdir()), newEnvironFrom(env)); err != nil {
-						logger.Fatal().Err(err).Msg("unable to execute command")
+					if cmd.Options != nil {
+						exc.CloudSyncDeployment = cmd.Options.CloudSyncDeployment
+						exc.CloudSyncTerraformPlanFile = cmd.Options.CloudSyncTerraformPlan
 					}
+
+					runs = append(runs, exc)
 				}
 			}
 		}
 	}
+
+	c.prepareScriptCloudDeploymentSync(runs)
+
+	isSuccessExit := func(exitCode int) bool {
+		return exitCode == 0
+	}
+
+	err := c.RunAll(runs, isSuccessExit, RunAllOptions{
+		Quiet:           c.parsedArgs.Quiet,
+		DryRun:          c.parsedArgs.Script.Run.DryRun,
+		ScriptRun:       true,
+		ContinueOnError: false,
+	})
+	if err != nil {
+		c.fatal("one or more commands failed", err)
+	}
 }
 
-func (c *cli) executeCommand(cmd []string, wd string, env []string) error {
+func (c *cli) prepareScriptCloudDeploymentSync(runStacks []RunContext) {
 	if c.parsedArgs.Script.Run.DryRun {
-		return nil
+		return
 	}
 
-	newCmd, err := makeCommand(cmd, wd, env, c.stdout, c.stderr)
-	if err != nil {
-		return errors.E(err, "failed to prepare command")
+	var deployRuns []RunContext
+	for _, exc := range runStacks {
+		if exc.CloudSyncDeployment {
+			deployRuns = append(deployRuns, exc)
+		}
 	}
 
-	if err := newCmd.Run(); err != nil {
-		return errors.E(err, "failed to execute command")
+	if len(deployRuns) == 0 {
+		return
 	}
 
-	return nil
-}
+	if !c.prj.isRepo {
+		fatal(errors.E("cloud features require a git repository"))
+	}
 
-func makeCommand(command []string, dir string, env []string, stdout, stderr io.Writer) (*exec.Cmd, error) {
-	cmdPath, err := run.LookPath(command[0], env)
-	if err != nil {
-		return nil, errors.E(err, "%s: command not found", command[0])
+	err := c.setupCloudConfig()
+	c.handleCriticalError(err)
+
+	if c.cloud.disabled {
+		return
```

Comment:
I think it would be more user friendly to print a warning here, and continue executing the command. As a user, I would prefer if my scripts continue to run irrespective of whether cloud is enabled or not.
What do you think?

Reply:
also, naked returns are a bit hard to read, e.g. the value of `err` returned here is not clear to me.

Reply:
Also more or less copied from how it's handled for `run`, in this case
`checkCloudSync()`.

I agree that it could be nicer though, but I'd say for the scope of this PR it's better to just stick to the current behavior of run, because it's already changing quite a few things.

Reply:
same comment as https://github.com/terramate-io/terramate/pull/1355#discussion_r1462990310